File: 백준/Silver/1149. RGB거리/RGB거리.py
import sys 
input = sys.stdin.readline
sys.setrecursionlimit(10**6)
n = int(input())
selected = [[0]*3 for _ in range(n)]
R=0
G=1
B=2
for i in range(n):
    r,g,b = map(int, input().split())
    if i==0:
        selected[0][R], selected[0][G], selected[0][B] =r,g,b
    else:
        selected[i][R] = min(selected[i-1][G], selected[i-1][B])+r  #R
        selected[i][G] = min(selected[i-1][R], selected[i-1][B])+g  #G
        selected[i][B] = min(selected[i-1][R], selected[i-1][G])+b  #B

print(min(selected[n-1]))

File: 백준/Silver/1149. RGB거리/RGB거리.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	var n int
	var r, g, b float64
	fmt.Fscanf(reader, "%d\n", &n)
	var selected = make([][]float64, n)
	for i := 0; i < n; i++ {
		selected[i] = make([]float64, 3)
	}
	for i := 0; i < n; i++ {
		fmt.Fscanf(reader, "%f %f %f\n", &r, &g, &b)
		if i == 0 {
			selected[0][0], selected[0][1], selected[0][2] = r, g, b
		} else {
			selected[i][0] = math.Min(selected[i-1][1], selected[i-1][2]) + r
			selected[i][1] = math.Min(selected[i-1][0], selected[i-1][2]) + g
			selected[i][2] = math.Min(selected[i-1][0], selected[i-1][1]) + b
		}
	}
	fmt.Println(math.Min(math.Min(selected[n-1][0], selected[n-1][1]), selected[n-1][2]))
}
